spec/requests/api/graphql/queries/consignments_spec.rb

```diff
@@ -0,0 +1,217 @@
+
+        consignments_response = body['data']['consignments']
+        expect(consignments_response).to eq nil
+
+        error_message = body['errors'][0]['message']
+
+        expect(error_message).to eq "Can't find partner."
+      end
+    end
+  end
+
+  describe 'valid requests' do
+    it 'returns the expected graphql response' do
+      post '/api/graphql', params: { query: query }, headers: headers
+
+      expect(response.status).to eq 200
+      body = JSON.parse(response.body)
+
+      expect(body['data']).to eq(
+        {
+          consignments: {
+            totalCount: 4,
+            edges: [
+              { node: { currency: 'USD', state: 'SOLD' } },
+              { node: { currency: 'USD', state: 'BOUGHT_IN' } },
+              { node: { currency: 'USD', state: 'SOLD' } },
+              { node: { currency: 'USD', state: 'BOUGHT_IN' } }
+            ]
+          }
+        }.deep_stringify_keys
+      )
+    end
+    it 'returns only sold or bought consignments' do
+      post '/api/graphql', params: { query: query }, headers: headers
+
+      expect(response.status).to eq 200
+      body = JSON.parse(response.body)
+
+      total_count = body.dig('data', 'consignments', 'totalCount')
+      state1 = body.dig('data', 'consignments', 'edges', 0, 'node', 'state')
+      state2 = body.dig('data', 'consignments', 'edges', 1, 'node', 'state')
+      state3 = body.dig('data', 'consignments', 'edges', 2, 'node', 'state')
+      state4 = body.dig('data', 'consignments', 'edges', 3, 'node', 'state')
+
+      expect(total_count).to eq 4
+      expect(state1).to eq 'SOLD'
+      expect(state2).to eq 'BOUGHT_IN'
+      expect(state3).to eq 'SOLD'
+      expect(state4).to eq 'BOUGHT_IN'
+    end
+
+    describe 'sorting' do
+      let(:query) do
+        <<-GRAPHQL
+        query {
+          consignments(#{query_inputs}) {
+             totalCount
+              edges {
+                node {
+                  currency
+                  id
+                  internalID
+                  saleDate
+                  saleName
+                  salePriceCents
+                  state
+                  submissionID
+                  submission {
+                    artistId
+                    category
+                    title
+                  }
+                }
+              }
+            }
+          }
+        GRAPHQL
+      end
+
+      context 'without a sort parameter' do
+        let(:query_inputs) do
+          "gravityPartnerId: \"#{partner.gravity_partner_id}\""
+        end
+
+        it 'returns the consignments sorted ascending by the id column' do
+          post '/api/graphql', params: { query: query }, headers: headers
+
+          expect(response.status).to eq 200
+          body = JSON.parse(response.body)
+
+          edges = body.dig('data', 'consignments', 'edges')
+          ids = edges.map { |edge| edge.dig('node', 'id') }
+
+          expect(ids).to eq %w[5 4 2 1]
```

Comment:
**praise:** nice specs!

Having just come off of a couple sprints worth of reading and writing sorting specs, I really appreciate how well this is done, thank you!! 🙏 

*format via [Conventional Comments](https://conventionalcomments.org/)*